src/main/scala/com/ocadotechnology/models/User.scala

```diff
@@ -0,0 +1,90 @@
+package com.ocadotechnology.models
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.string._
+import eu.timepit.refined.types.string._
+import eu.timepit.refined.types.numeric._
+
+import io.circe.Codec // Codec type
+import io.circe.generic.semiauto._ // Derivation of codecs
+import io.circe.refined._ // Derivations for refined types
+import io.circe.syntax._ // for `.toJson` syntax
+import io.circe.parser._ // for parsing json
+
+
+type Email = String Refined MatchesRegex["""[a-z0-9]+@[a-z0-9]+\\.[a-z0-9]{2,}"""]
+
+/*
+Some unexpected behaviour spotted: when building the project, it sometimes produces error:
+
+
+Could not find Schema for type String Refined
+  eu.timepit.refined.string.MatchesRegex[
+    ("[a-z0-9]+@[a-z0-9]+\\\\.[a-z0-9]{2,}" : String)
+  ].
+Schemas can be derived automatically by adding: `import sttp.tapir.generic.auto._`, or manually using `Schema.derived[T]`.
+The latter is also useful for debugging derivation errors.
+You can find more details in the docs: https://tapir.softwaremill.com/en/latest/endpoint/schemas.html
+When integrating with a third-party datatype library remember to import respective schemas/codecs as described in https://tapir.softwaremill.com/en/latest/endpoint/integrations.html.
+I found:
+
+    sttp.tapir.Schema.derivedEnumerationValue[Enumeration#Value]
+
+But method derivedEnumerationValue in trait SchemaCompanionMacros does not match type sttp.tapir.Schema[String Refined
+  eu.timepit.refined.string.MatchesRegex[
+    ("[a-z0-9]+@[a-z0-9]+\\\\.[a-z0-9]{2,}" : String)
+  ]
+].
+
+One of the following imports might make progress towards fixing the problem:
+
+  import shapeless.~?>.idKeyWitness
+  import shapeless.~?>.idValueWitness
+  import shapeless.~?>.witness
+
+
+    .out(jsonBody[User])
+
+
+
+
+The only working solution I found is to switch back to generic types, build the project and then switch to refined types and build the project again
+Turns out that every change in Endpoints.scala (even adding a new line) results in throwing this error.
+*/
```

Comment:
> The only working solution I found is to switch back to generic types, build the project and then switch to refined types and build the project again

So at this point you basically rely on the binary version of the file, as long as it was compiled once, and the data type was valid at that point it would work on partial compilation I suppose :wink:  Nevertheless I appreciate the effort, it's an interesting finding on its own!

The solution here is very simple, just accept the suggestion at https://github.com/ocadotechnology-university/doubloons/pull/2/files#r1148822552 

It's actually in the error message, it says:

> When integrating with a third-party datatype library remember to import respective schemas/codecs as described in https://tapir.softwaremill.com/en/latest/endpoint/integrations.html.

Now of course this might not sound obvious at the first sight, but here we rely on the tapir-refined integration, it's in the `build.sbt` file
```scala
"com.softwaremill.sttp.tapir" %% "tapir-refined" % Versions.tapir,
```
In the [datatype integrations](https://tapir.softwaremill.com/en/latest/endpoint/integrations.html) document mentioned in the error, you'll find the following section about refined:

![tapir-integration-docs](https://user-images.githubusercontent.com/7260594/227855618-306f8ba7-4f1a-4d5e-8cf5-c1bfa9a37b3e.png)

So the solution is to simply add the import:
```scala
import sttp.tapir.codec.refined.*
```

This is required because Tapir generates implicit `Schema` objects, that represents a bunch of metadata about the endpoint, including the codecs and endpoint validation rules. When all goes well, you shouldn't have to worry about it, and the endpoint simply summons some extra implicits to the context to make sure Tapir is able to generate schemas that include refined types.

You can learn more about them in https://tapir.softwaremill.com/en/latest/endpoint/codecs.html#schemas

Reply:
You are right. I added the tapir-refined module to the dependencies, but didn't manage to use the correct import statement.
Thank You for the feedback!